dict.c

```diff
@@ -210,16 +210,11 @@ static dictEntry *dictFind(dict *ht, const void *key) {
     return NULL;
 }
 
-static dictIterator *dictGetIterator(dict *ht) {
-    dictIterator *iter = hi_malloc(sizeof(*iter));
-    if (iter == NULL)
-        return NULL;
-
+static void dictInitIterator(dictIterator *iter, dict *ht) {
```

Comment:
This means we're diverging from the dict.c originally copied from Redis?

Reply:
There is a dict in Redis, and one in Hiredis that probably originate from Redis, and they differs a lot today.
Ours is originated from Hiredis, and this will make it diverge.
Since the dictGetIter.. and dictReleaseIter.. was no longer used they triggered build warnings.
Maybe its possible to try to introduced the dictInitIterator to the hiredis repo, then use the dict from hiredis as a dependency..

Reply:
OK, I just think it might be good to keep track of at least the one from Redis. If we pick code from Redis, it will make it diverge less, rather than more. I see Redis has `dictGetIterator()` which allocates the iterator on the heap and no `dictInitIterator()`... But maybe we can introduce it there too in the future(?).